crates/aptos-crypto/src/p256/p256_keys.rs

```diff
@@ -0,0 +1,270 @@
+
+impl P256PrivateKey {
+    /// The length of the P256PrivateKey
+    pub const LENGTH: usize = P256_PRIVATE_KEY_LENGTH;
+
+    /// Serialize a P256PrivateKey.
+    pub fn to_bytes(&self) -> [u8; P256_PRIVATE_KEY_LENGTH] {
+        self.0.to_bytes().into()
+    }
+
+    /// Deserialize an P256PrivateKey without any validation checks apart from expected key size.
+    fn from_bytes_unchecked(
+        bytes: &[u8],
+    ) -> std::result::Result<P256PrivateKey, CryptoMaterialError> {
+        match p256::ecdsa::SigningKey::from_slice(bytes) {
+            Ok(p256_secret_key) => Ok(P256PrivateKey(p256_secret_key)),
+            Err(_) => Err(CryptoMaterialError::DeserializationError),
+        }
+    }
+
+    /// Private function aimed at minimizing code duplication between sign
+    /// methods of the SigningKey implementation. This should remain private.
+    fn sign_arbitrary_message(&self, message: &[u8]) -> P256Signature {
+        let secret_key: &p256::ecdsa::SigningKey = &self.0;
+        let sig = P256Signature(secret_key.sign(message.as_ref()));
+        let canonical_sig = P256Signature::make_canonical(&sig);
+
+        canonical_sig
+    }
+}
+
+impl P256PublicKey {
+    /// Serialize a P256PublicKey.
+    pub fn to_bytes(&self) -> [u8; P256_PUBLIC_KEY_LENGTH] {
+        // The RustCrypto P256 `to_sec1_bytes` call here should never return an array of the wrong length and cause a panic
+        (*self.0.to_sec1_bytes()).try_into().unwrap()
+    }
+
+    /// Deserialize a P256PublicKey, checking expected key size
+    /// and that it is a valid curve point.
+    pub(crate) fn from_bytes_unchecked(
+        bytes: &[u8],
+    ) -> std::result::Result<P256PublicKey, CryptoMaterialError> {
+        match p256::ecdsa::VerifyingKey::from_sec1_bytes(bytes) {
+            Ok(p256_public_key) => Ok(P256PublicKey(p256_public_key)),
+            Err(_) => Err(CryptoMaterialError::DeserializationError),
+        }
+    }
+}
+
+///////////////////////
+// PrivateKey Traits //
+///////////////////////
+
+impl PrivateKey for P256PrivateKey {
+    type PublicKeyMaterial = P256PublicKey;
+}
+
+impl SigningKey for P256PrivateKey {
+    type SignatureMaterial = P256Signature;
+    type VerifyingKeyMaterial = P256PublicKey;
+
+    fn sign<T: CryptoHash + Serialize>(
+        &self,
+        message: &T,
+    ) -> Result<P256Signature, CryptoMaterialError> {
+        Ok(P256PrivateKey::sign_arbitrary_message(
+            self,
+            signing_message(message)?.as_ref(),
+        ))
+    }
+
+    #[cfg(any(test, feature = "fuzzing"))]
+    fn sign_arbitrary_message(&self, message: &[u8]) -> P256Signature {
+        P256PrivateKey::sign_arbitrary_message(self, message)
+    }
+}
+
+impl Uniform for P256PrivateKey {
+    fn generate<R>(rng: &mut R) -> Self
+    where
+       R: ::rand::RngCore + ::rand::CryptoRng + ::rand_core::CryptoRng + ::rand_core::RngCore,
+    {
+        let mut bytes: [u8; P256_PRIVATE_KEY_LENGTH] = Default::default();
```

Comment:
You do not need the `Default` call.

Reply:
`let mut bytes: [0u8; P256_PRIVATE_KEY_LENGTH];` gives a compiler error

Reply:
Ugh. Because `:` should be a `=`...